### PR TITLE
Squash v1, v2, v3 schemas into single branch

### PR DESCRIPTION
**Pull request by wagoodman**
This PR does the following:
- merges the v1, v2, v3 branches into a single branch
- removes the `db.store` abstraction
- migrates namespace determination into the appropriate `v*` package. These are existing functions from `grype` and `grype-db-builder`

Do before merging:
- [x] patch back in https://github.com/anchore/grype-db/pull/30 into the latest changes

### CODE REVIEW

File: pkg/db/v3/namespace.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
+package v3
```
Comment (luhring) [inline pkg/db/v3/namespace.go:1]: (not a request destined for this PR, just noting a conversation) We discussed a potential improvement for the future, where the Grype-specific and Builder-specific concerns would be moved to those respective codebases, and what would be provided by Grype-DB would be squarely the data types, values, and simple value-producing functions that are common to both production and consumption of the database. In doing so, this code would be oblivious to changes in Grype needs and Builder needs.

("Builder" == grype-db-builder)

File: pkg/db/v3/id.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package v3
+
+import (
+	"time"
+)
+
+type ID struct {
```
Comment (luhring) [inline pkg/db/v3/id.go:7]: nit: It'd be helpful to document this type to explain how it's intended to be used. (For example, to answer questions like "Is this an ID that can be used for _any_ data records?")

**Review by luhring — APPROVED** [review]
Very nice work!

File: pkg/db/v3/id.go (hunk 2 above)
Comment (wagoodman) [reply pkg/db/v3/id.go:8]: added comments for ID in v1, v2, and v3 as well as a few other structs and fields.

**Pull request merged**